### ISSUE TITLE
generation of output video fails with: Could not find tag for codec pcm_s16le in stream #1, codec not currently supported in container

### ISSUE THREAD

**Issue by carandraug** [ox-vgg/chimpanzee-tracking#1]
Reported by Dan. Running in Colab, the final step that generates a video showing the detections, fails with:

```
ffmpeg version 4.2.7-0ubuntu0.1 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
  configuration: --prefix=/usr --extra-version=0ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Input #0, image2, from '/tmp/tmpf07oeg39/*.jpg':
  Duration: 00:00:30.23, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: mjpeg (Baseline), yuvj420p(pc, bt470bg/unknown/unknown), 1920x1080 [SAR 1:1 DAR 16:9], 30 fps, 30 tbr, 30 tbn, 30 tbc
Guessed Channel Layout for Input Stream #1.1 : mono
Input #1, mov,mp4,m4a,3gp,3g2,mj2, from '/content/drive/MyDrive/video-data/camera-trap/chimp-detection.MP4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp41
    creation_time   : 2022-04-11T16:04:13.000000Z
  Duration: 00:01:00.50, start: 0.000000, bitrate: 12804 kb/s
    Stream #1:0(eng): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1920x1080, 12095 kb/s, 30 fps, 30 tbr, 60k tbn, 120k tbc (default)
    Metadata:
      creation_time   : 2022-04-11T16:04:13.000000Z
      handler_name    : Video Handler
      encoder         : iCatch AVCC
    Stream #1:1(eng): Audio: pcm_s16le (sowt / 0x74776F73), 44100 Hz, mono, s16, 705 kb/s (default)
    Metadata:
      creation_time   : 2022-04-11T16:04:13.000000Z
      handler_name    : Audio Handler
Stream mapping:
  Stream #0:0 -> #0:0 (mjpeg (native) -> h264 (libx264))
  Stream #1:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
[swscaler @ 0x560b98c33240] deprecated pixel format used, make sure you did set range correctly
[libx264 @ 0x560b9927d800] using SAR=1/1
[libx264 @ 0x560b9927d800] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2 AVX512
[libx264 @ 0x560b9927d800] profile High, level 4.0
[libx264 @ 0x560b9927d800] 264 - core 155 r2917 0a84d98 - H.264/MPEG-4 AVC codec - Copyleft 2003-2018 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=3 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=25 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00
[mp4 @ 0x560b98bd2900] Could not find tag for codec pcm_s16le in stream #1, codec not currently supported in container
Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument
Error initializing output stream 0:0 --
[image2 @ 0x560b98bb9140] Thread message queue blocking; consider raising the thread_queue_size option (current value: 8)
Conversion failed!
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-15-3886b39a245b> in <cell line: 5>()
      6     tmp_tracks_fpath = os.path.join(out_frames_dir, 'tracks.mp4')
      7     make_frames_with_tracks(CSV_FPATH, FRAMES_DIR, out_frames_dir)
----> 8     ffmpeg_video_from_frames_and_video(
      9         out_frames_dir, VIDEO_FILE, tmp_tracks_fpath
     10     )1 frames
<ipython-input-10-6f9bb5f46137> in subprocess_print_stderr(args)
     72     p.wait()
     73     if p.returncode != 0:
---> 74         raise Exception('subprocess failed')
     75
     76Exception: subprocess failed
```

Google Drive link to obtain the video that causes issues:
https://drive.google.com/file/d/1TNsFCQJiaNVv8CDsXKMGOB60AjQFxv-2/view?usp=sharing_eip_m&ts=6442b864

**Comment by carandraug** [comment]
To replicate locally in Debian 10:

    # create video to frames
    mkdir frames/
    ffmpeg -i chimp-detection.MP4 -vsync vfr -q:v 1 -start_number 0 -filter:v "scale=iw:ih*(1/sar)" "frames/%06d.jpg"

    # frames to video (errors as in the original post)
    ffmpeg -framerate 30 -pattern_type glob -i 'frames/*.jpg' -i chimp-detection.MP4 -c:a copy -c:v libx264 -map 0:v:0 -map 1:a:0 -pix_fmt yuv420p tt.mp4

The root problem is that ffmpeg does not support muxing PCM audio in mp4. The reason why, is that PCM audio in mp4 is not even part of the mp4 specification. While ffmpeg supports reading of these, apparently incorrect, mp4 files it does not support writing them (see discussion in [related ffmpeg issue ](https://trac.ffmpeg.org/ticket/3818)). On that note, it seems that pcm audio in mp4 files has been added to the mp4 spec very recently so maybe future versions of ffmpeg will support it.

Anyway, the simplest solution is to just decode and re-encode the audio with a code that ffmpeg will support for sure (acc is the default and seems to work fine). This will be a bit slower but at least works. It can also introduce artefacts in the sound but again, at least works, and the we're only working with video and doing this to output an video showing the detections so should be very much acceptable to lose sound quality (we're losing much more in image quality and image is the thing we care about).

**Comment by carandraug** [comment]
Fixed in 4fb39ce by forcing aac audio codec for output. Closing as fixed.
